corehq/apps/hqcase/management/commands/ptop_preindex.py

```diff
@@ -33,42 +43,49 @@
 
 
 def get_reindex_commands(hq_index_name):
-    # pillow_command_map is a mapping from es pillows
-    # to lists of management commands or functions
-    # that should be used to rebuild the index from scratch
+    """Return a list of ``ReindexerFactory`` classes or functions that are used
+    to rebuild the index from scratch.
+
+    :param hq_index_name: ``str`` name of the Elastic index alias"""
     pillow_command_map = {
-        DOMAIN_HQ_INDEX_NAME: ['domain'],
-        CASE_HQ_INDEX_NAME: ['case', 'sql-case'],
-        XFORM_HQ_INDEX_NAME: ['sql-form'],
+        DOMAIN_HQ_INDEX_NAME: [DomainReindexerFactory],
+        CASE_HQ_INDEX_NAME: [SqlCaseReindexerFactory],
+        XFORM_HQ_INDEX_NAME: [SqlFormReindexerFactory],
         # groupstousers indexing must happen after all users are indexed
         USER_HQ_INDEX_NAME: [
-            'user',
+            UserReindexerFactory,
             add_demo_user_to_user_index,
-            'groups-to-user',
+            GroupToUserReindexerFactory,
         ],
-        APP_HQ_INDEX_NAME: ['app'],
-        GROUP_HQ_INDEX_NAME: ['group'],
-        REPORT_XFORM_HQ_INDEX_NAME: ['report-xform'],
-        REPORT_CASE_HQ_INDEX_NAME: ['report-case'],
-        CASE_SEARCH_HQ_INDEX_NAME: ['case-search'],
-        SMS_HQ_INDEX_NAME: ['sms'],
+        APP_HQ_INDEX_NAME: [AppReindexerFactory],
+        GROUP_HQ_INDEX_NAME: [GroupReindexerFactory],
+        REPORT_XFORM_HQ_INDEX_NAME: [ReportFormReindexerFactory],
+        REPORT_CASE_HQ_INDEX_NAME: [ReportCaseReindexerFactory],
+        CASE_SEARCH_HQ_INDEX_NAME: [CaseSearchReindexerFactory],
+        SMS_HQ_INDEX_NAME: [SmsReindexerFactory],
     }
     return pillow_command_map.get(hq_index_name, [])
 
 
 def do_reindex(hq_index_name, reset):
     print("Starting pillow preindex %s" % hq_index_name)
     reindex_commands = get_reindex_commands(hq_index_name)
-    for reindex_command in reindex_commands:
-        if isinstance(reindex_command, str):
-            kwargs = {}
-            factory = FACTORIES_BY_SLUG[reindex_command]
-            reindex_args = ReindexerFactory.resumable_reindexer_args
-            if reset and factory.arg_contributors and reindex_args in factory.arg_contributors:
-                kwargs["reset"] = True
-            factory(**kwargs).build().reindex()
+    for factory_or_func in reindex_commands:
+        try:
+            is_factory = issubclass(factory_or_func, ReindexerFactory)
+        except TypeError:  # TypeError: issubclass() arg 1 must be a class
+            factory_or_func()
```

Comment:
I find this `try`/`except` a bit disorienting because the exception is actually the more common case (it's not exceptional). What do you think of this?
```suggestion
        if not isinstance(factory_or_func, type):
            factory_or_func()
        elif not issubclass(factory_or_func, ReindexerFactory):
            raise ValueError(f"expected ReindexerFactory, got: {factory_or_func!r}")
```

This is not a pure suggestion (can't be applied safely) because the code below now references an undefined variable `is_factory`. In fact, I'd change it to state the if conditions in a positive way:
```py
        if issubclass(factory_or_func, ReindexerFactory):
            kwargs = {}
            ...
            factory_or_func(**kwargs).build().reindex()
        elif isinstance(factory_or_func, type):
            raise ValueError(f"expected ReindexerFactory, got: {factory_or_func!r}")
        else:
            factory_or_func()
```